controllers/argocd/util.go

```diff
@@ -1688,3 +1688,31 @@ func UseServer(name string, cr *argoproj.ArgoCD) bool {
 	}
 	return true
 }
+
+// addKubernetesData checks for any Kubernetes-specific labels or annotations
+// in the live object and updates the source object to ensure critical metadata
+// (like scheduling, topology, or lifecycle information) is retained.
+// This helps avoid loss of important Kubernetes-managed metadata during updates.
+func addKubernetesData(source map[string]string, live map[string]string) map[string]string {
+
+	// List of Kubernetes-specific substrings (wildcard match)
+	patterns := []string{
+		"*kubernetes.io*",
+		"*k8s.io*",
+		"*openshift.io*",
+	}
+
+	for key, value := range live {
+		found := glob.MatchStringInList(patterns, key, glob.GLOB)
+		if found {
+			// Don't override values already present in the source object.
+			// This ensures users have control over Kubernetes-managed data
+			// if they have intentionally set or modified these values in the source object.
+			if _, ok := source[key]; !ok {
+				source[key] = value
+			}
+		}
```

Comment:
QQ: Will live object have the latest data for k8s annotations/labels? If yes, should we directly update the value from live object into source instead of checking it to be present?

Reply:
Live will always have latest data for k8s annotations/labels. The check is to ensure, users can update the value from ArgoCD CR if required. Example: by setting
```yaml
kind: ArgoCD
spec:
  server:
    annotations:
      node.kubernetes.io/pod: new-value
``` 

Without this check users cannot modify kuberentes/openshift label/annotations if they wish to.

Reply:
Would kubernetes not overwrite those values again given that the values are generated by kubernetes?

Reply:
Okay. When I think of it again, I feel the logic is required because operator also sets some kubernetes.io & openshift.io labels & annotations. If the logic is not present, any update operation on these labels/annots from operator won't reconcile. They will always take live value which will be the old one.

Reply:
In that case, we can continue with the logic you had added.